### PR TITLE
Add mutation for bulk delete order lines

### PR DESCRIPTION
**Pull request by akjanik**
Part of  #3927

### Pull Request Checklist

1. [x] Privileged views and APIs are guarded by proper permission checks.
1. [x] All visible strings are translated with proper context.
1. [x] All data-formatting is locale-aware (dates, numbers, and so on).
1. [x] Database queries are optimized and the number of queries is constant.
1. [x] Database migration files are up to date.
1. [x] The changes are tested.
1. [x] The code is documented (docstrings, project documentation).
1. [x] GraphQL schema and type definitions are up to date.
1. [x] Changes are mentioned in the changelog.

### CODE REVIEW

File: tests/api/test_bulk_delete.py [hunk 1]
```diff
@@ -286,6 +286,46 @@ def test_delete_draft_orders(
         id__in=[order.id for order in orders]).count() == len(orders)
 
 
+def test_delete_draft_orders_lines(
+        staff_api_client, order_with_lines, permission_manage_orders):
+    order = order_with_lines
+    order_lines = [line for line in order]
+    order.status = OrderStatus.CANCELED
```
Comment (korycins) [inline tests/api/test_bulk_delete.py]: Do you need this field?
I don't see a place where you are saving this change to DB

File: tests/api/test_bulk_delete.py [hunk 2]
```diff
@@ -286,6 +286,46 @@ def test_delete_draft_orders(
...
+        graphene.Node.to_global_id('OrderLine', order_line.id)
+        for order_line in order_lines]}
+    response = staff_api_client.post_graphql(
+        query, variables,
+        permissions=[permission_manage_orders])
+    content = get_graphql_content(response)
+    assert 'errors' in content['data']['draftOrderLinesBulkDelete']
+    assert content['data']['draftOrderLinesBulkDelete']['count'] == 0
+
+    order.status = OrderStatus.DRAFT
```
Comment (korycins) [inline tests/api/test_bulk_delete.py]: I think that part of this unittest, starting from here should be in another one. 
You are testing two different things. 1. The behaviour for non-draft orders 2. The behaviour for draft orders.

Comment (akjanik) [reply tests/api/test_bulk_delete.py]: Yep, agree. I just wanted to test draft and non draft order cases in one test, need to divide in two.

**Comment by akjanik** [comment]
I am putting this on hold, untill I finish #3891

**Comment by korycins** [comment]
@NyanKiyoshi, @akjanik Are we ready to merge it?

**Comment by maarcingebala** [comment]
What about other bulk actions that you wanted to add @akjanik? Are we going to do them in separate PR (which is perfectly fine, just asking out of curiosity to know what's our plan)? Also, I don't know if you have it on your list, but we'll need `bulkCancelOrder` mutation to be able to cancel multiple orders (as an open, non-draft order cannot be deleted).

**Comment by akjanik** [comment]
@maarcingebala To avoid issues with large PRs, I decided to divide work into small ones. ATM I am working on `BulkPublishPages`, but after that, I may focus on this bulk cancel mutation. If more missing mutations comes to your mind, feel free to mention them in #3927

File: tests/api/test_bulk_delete.py [hunk 3]
```diff
@@ -286,6 +298,48 @@ def test_delete_draft_orders(
         id__in=[order.id for order in orders]).count() == len(orders)
 
 
+def test_fail_to_delete_non_draft_order_lines(
+        staff_api_client, order_with_lines, permission_manage_orders):
+    order = order_with_lines
+    order_lines = [line for line in order]
+    # Set anything but OrderStatus.DRAFT
```
Comment (NyanKiyoshi) [inline tests/api/test_bulk_delete.py]:
```suggestion
    # Ensure we cannot delete a non-draft order
```

File: tests/api/test_bulk_delete.py [hunk 4]
```diff
@@ -286,6 +298,48 @@ def test_delete_draft_orders(
         id__in=[order.id for order in orders]).count() == len(orders)
 
 
+def test_fail_to_delete_non_draft_order_lines(
+        staff_api_client, order_with_lines, permission_manage_orders):
+    order = order_with_lines
+    order_lines = [line for line in order]
```
Comment (NyanKiyoshi) [inline tests/api/test_bulk_delete.py]: Wouldn't that work to do `order.lines`?

File: tests/api/test_bulk_delete.py [hunk 5]
```diff
@@ -286,6 +298,48 @@ def test_delete_draft_orders(
...
+    # Set anything but OrderStatus.DRAFT
+    order.status = OrderStatus.CANCELED
+    order.save()
+
+    variables = {'ids': [
+        graphene.Node.to_global_id('OrderLine', order_line.id)
+        for order_line in order_lines]}
+    response = staff_api_client.post_graphql(
```
Comment (NyanKiyoshi) [inline tests/api/test_bulk_delete.py]:
```suggestion

    # Post the invalid request and 
    # ensure the correct permissions are required
    response = staff_api_client.post_graphql(
```

File: tests/api/test_bulk_delete.py [hunk 6]
```diff
@@ -286,6 +298,48 @@ def test_delete_draft_orders(
...
+
+    content = get_graphql_content(response)
+    assert 'errors' in content['data']['draftOrderLinesBulkDelete']
+    assert content['data']['draftOrderLinesBulkDelete']['count'] == 0
+
+
+def test_delete_draft_order_lines(
+        staff_api_client, order_with_lines, permission_manage_orders):
+    order = order_with_lines
+    order_lines = [line for line in order]
```
Comment (NyanKiyoshi) [inline tests/api/test_bulk_delete.py]: `order.lines`?

Comment (akjanik) [reply tests/api/test_bulk_delete.py]: I was recently caught on this too, but `Order` has overridden method:
```
    def __iter__(self):
        return iter(self.lines.all())
```

File: tests/api/test_bulk_delete.py (hunk 5 above)
Comment (akjanik) [reply tests/api/test_bulk_delete.py]: I don't think this comment is necessary, as we are doing it for almost every request that requires permission(s).

As a side note, I would like to note, it slows us down IMO, as we are doing 2 requests, one which checks that user has no permission and second, that adds the appropriate one. It is tedious to use too. Maybe we can talk with @maarcingebala about this in the near future.

**Pull request merged**